### PR TITLE
Modify GitHub data generator to work with GitHub app credentials

### PR DESCRIPTION
**Pull request by wbssbw**
This makes the GitHub data generator compatible with GitHub app creds, removing the need for FPATs as described in [this issue](https://github.com/obelisk/plaid/issues/6)

It could be nice to have the API and data generator share the same GitHub client, but that would require breaking changes to the config to move the authentication settings outside of the API and data generators section.

### CODE REVIEW

**Review by obelisk — COMMENTED** [review]
This looks good. Can you update the example plaid.toml so I can easily see how this is configured?

File: plaid/src/data/github/mod.rs [hunk 1]
```diff
@@ -55,97 +63,140 @@ impl Ord for GithubAuditLog {
...
+    D: serde::Deserializer<'de>,
+{
+    let log_type = String::deserialize(deserializer)?.to_lowercase();
+
+    match log_type.as_str() {
+        "web" => Ok(LogType::Web),
+        "git" => Ok(LogType::Git),
+        "all" => Ok(LogType::All),
+        _ => Err(serde::de::Error::custom(
+            "Invalid log tpye provided. Expected one of All, Git, Web",
```
Comment (obelisk) [inline plaid/src/data/github/mod.rs:104]: Nit typo

**Review by obelisk — APPROVED** [review]

**Comment by obelisk** [comment]
We are going to need to test this heavily but otherwise looks good

File: plaid/src/data/github/mod.rs (hunk 1 above)
Comment (wbssbw) [reply plaid/src/data/github/mod.rs]: Fixed!

**Review by obelisk — APPROVED** [review]

**Pull request merged**